doc/developer/design/20210622_progress_observation.md

```diff
@@ -0,0 +1,104 @@
+# Observe source progress
+
+## Summary
+
+Currently, Materialize does not provide any feedback about the source
+progression. This makes it difficult to determine if a source is up-to-date and
+queries return fresh data. The goal of this effort is to add a mechanism to
+allow clients to query the progress of a source in a provider-independent way.
+
+## Goals
+
+The goals of this design are to enable users to observe the progress of their
+sources. It should help answer the following questions:
+
+* Is all (past) data loaded?
+* What was the last time the source was up-to-date?
+* How much does a source lag behind?
+
+## Non-Goals
+
+A non-goal is to load data in a more efficient manner.
```

Comment:
As mentioned above, I think the following should be listed here:
* What was the last time the source was up-to-date?
* How much does a source lag behind?